ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveSubQueryRemoveRule.java

```diff
@@ -105,11 +105,9 @@ public void onMatch(RelOptRuleCall call) {
       SubqueryConf subqueryConfig = filter.getCluster().getPlanner().
           getContext().unwrap(SubqueryConf.class);
       boolean isCorrScalarQuery = subqueryConfig.getCorrScalarRexSQWithAgg().contains(e.rel);
-      boolean hasNoWindowingAndNoGby =
-          subqueryConfig.getScalarAggWithoutGbyWindowing().contains(e.rel);
```

Comment:
With this patch ```getScalarAggWithoutGbyWindowing``` does not seem to be used anymore. If that is the case, we should remove it from the config, from the context, and get rid of it in ```CalcitePlanner``` too.